src/ios/AppDelegate+notification.m

```diff
@@ -62,7 +62,7 @@ - (void)application:(UIApplication *)application didFailToRegisterForRemoteNotif
     [pushHandler didFailToRegisterForRemoteNotificationsWithError:error];
 }
 
-- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
+- (void)applicationPushPlugin:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
```

Comment:
After looking at the Cordova source code, I think I understand why PushPlugin implements didReceiveRemoteNotification.

Cordova implemented an event for when you register for remote notifications.  However, they did not implement an event for when you *receive* notifications.  I don't know why they didn't implement this event.  They implemented this event for local notifications, so maybe they just forgot for remote notifications?!

[Remote notifications implementation](https://github.com/apache/cordova-ios/blob/master/bin/templates/project/__PROJECT_NAME__/Classes/AppDelegate.m#L116)
```
#ifndef DISABLE_PUSH_NOTIFICATIONS

    - (void)                                 application:(UIApplication*)application
        didRegisterForRemoteNotificationsWithDeviceToken:(NSData*)deviceToken
    {
        // re-post ( broadcast )
        NSString* token = [[[[deviceToken description]
            stringByReplacingOccurrencesOfString:@"<" withString:@""]
            stringByReplacingOccurrencesOfString:@">" withString:@""]
            stringByReplacingOccurrencesOfString:@" " withString:@""];

        [[NSNotificationCenter defaultCenter] postNotificationName:CDVRemoteNotification object:token];
    }

    - (void)                                 application:(UIApplication*)application
        didFailToRegisterForRemoteNotificationsWithError:(NSError*)error
    {
        // re-post ( broadcast )
        [[NSNotificationCenter defaultCenter] postNotificationName:CDVRemoteNotificationError object:error];
    }
#endif
```

[Local notifications implementation](https://github.com/apache/cordova-ios/blob/master/bin/templates/project/__PROJECT_NAME__/Classes/AppDelegate.m#L108)
```
// repost all remote and local notification using the default NSNotificationCenter so multiple plugins may respond
- (void)            application:(UIApplication*)application
    didReceiveLocalNotification:(UILocalNotification*)notification
{
    // re-post ( broadcast )
    [[NSNotificationCenter defaultCenter] postNotificationName:CDVLocalNotification object:notification];
}
```

Notice there is no implementation for didReceiveRemoteNotification.  I thought something like the following would exist, but it does not.
```
 (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
    [[NSNotificationCenter defaultCenter] postNotificationName:CDVRemoteNotificationReceived object:userInfo];
}
```